From 1766f2baf6cf3e938fe473d6ffabf4633ce5bc10 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 31 Oct 2012 23:38:54 -0400
Subject: [PATCH 0035/1936] UI: Misc fixes with respect to NTK support

- Permits running program without installing pixmaps by referencing them in the
  source directory when not found in the install path
- FLTK build uses FL_PLASTIC_UP_BOX rather than FL_UP_BOX
- Blank space eliminated from MasterUI.fl
- Effect preset selection uses a more visible font color
- Padsynth's harmonic view is clearer with a more subtle grid
- Updates AUTHORS.txt file
---
 AUTHORS.txt           |   2 +-
 src/CMakeLists.txt    |   3 +-
 src/UI/CMakeLists.txt |   1 +
 src/UI/EffUI.fl       | 308 +++++++++++++++++++++---------------------
 src/UI/MasterUI.fl    |  18 +--
 src/UI/PADnoteUI.fl   |   4 +-
 src/UI/VirKeyboard.fl |   9 ++
 src/main.cpp          |  23 +++-
 8 files changed, 195 insertions(+), 173 deletions(-)

diff --git a/AUTHORS.txt b/AUTHORS.txt
index 1ddccc12..14d2cade 100644
--- a/AUTHORS.txt
+++ b/AUTHORS.txt
@@ -22,5 +22,5 @@ Contributors:
         Damien Goutte-Gattat (Bank select midi support)
         Lieven Moors (Spike/Circle waveform)
         Olaf Schulz (MIDI Aftertouch support)
-        Jonathan Liles (NSM support)
+        Jonathan Liles (NSM & NTK support)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3c7cddf2..8de9d098 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -32,7 +32,7 @@ mark_as_advanced(LIBLO_LIBRARIES)
 # NOTE: These cache variables should normally not be changed in this
 # file, but either in in CMakeCache.txt before compile, or by passing
 # parameters directly into cmake using the -D flag.
-SET (GuiModule fltk CACHE STRING "GUI module, either fltk, qt or off")
+SET (GuiModule fltk CACHE STRING "GUI module, either fltk, ntk or off")
 SET (CompileTests ${CXXTEST_FOUND} CACHE BOOL "whether tests should be compiled in or not")
 SET (AlsaEnable ${ALSA_FOUND} CACHE BOOL
     "Enable support for Advanced Linux Sound Architecture")
@@ -366,6 +366,7 @@ install(TARGETS zynaddsubfx
 if(NtkGui)
 install(DIRECTORY ../pixmaps DESTINATION share/zynaddsubfx)
 add_definitions(-DPIXMAP_PATH="${CMAKE_INSTALL_PREFIX}/share/zynaddsubfx/pixmaps/")
+add_definitions(-DSOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}")
 endif(NtkGui)
 
 include(CTest)
diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index 16e5b6a6..38ea1e11 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -23,6 +23,7 @@ set_source_files_properties(UI/MasterUI.h PROPERTIES GENERATED 1)
 fltk_wrap_ui(zynaddsubfx_gui ${UI_fl_files})
 
 add_definitions(-DPIXMAP_PATH="${CMAKE_INSTALL_PREFIX}/share/zynaddsubfx/pixmaps/")
+add_definitions(-DSOURCE_DIR="${CMAKE_CURRENT_SOURCE_DIR}")
 
 if(LibloEnable)
     set(zynaddsubfx_gui_FLTK_UI_SRCS ${zynaddsubfx_gui_FLTK_UI_SRCS} NSM.C NSM/Client.C)
diff --git a/src/UI/EffUI.fl b/src/UI/EffUI.fl
index 76bc03a0..762aa4d0 100644
--- a/src/UI/EffUI.fl
+++ b/src/UI/EffUI.fl
@@ -194,7 +194,7 @@ if (filterwindow!=NULL){
   } {
     Fl_Window effnullwindow {
       label {No Effect}
-      xywh {857 824 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 16
+      xywh {612 881 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 16
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {}
@@ -202,8 +202,8 @@ if (filterwindow!=NULL){
   Function {make_reverb_window()} {open
   } {
     Fl_Window effreverbwindow {
-      label Reverb
-      xywh {353 452 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      label Reverb open
+      xywh {377 636 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -212,59 +212,59 @@ if (filterwindow!=NULL){
         callback {eff->changepreset((int)o->value());
 
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Cathedral 1}
-          xywh {10 10 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {10 10 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Cathedral 2}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Cathedral 3}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Hall 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Hall 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Room 1}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Room 2}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Basement
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Tunnel
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echoed 1}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echoed 2}
-          xywh {110 110 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {110 110 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Very Long 1}
-          xywh {120 120 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {120 120 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Very Long 2}
-          xywh {130 130 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {130 130 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Choice revp10 {
@@ -272,19 +272,19 @@ refresh(eff);}
         callback {eff->seteffectpar(10,(int) o->value());
 if (eff->geteffectpar(10)==2) revp12->activate();
 	else revp12->deactivate();}
-        xywh {110 15 85 15} down_box BORDER_BOX color 14 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {110 15 85 15} down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label Random
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Freeverb
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Bandwidth
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial revp0 {
@@ -363,7 +363,7 @@ eff->seteffectpar(11,x);}
   } {
     Fl_Window effechowindow {
       label Echo
-      xywh {1101 647 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      xywh {897 611 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -371,43 +371,43 @@ eff->seteffectpar(11,x);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {11 15 95 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {11 15 95 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Echo 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echo 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echo 3}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Simple Echo}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Canyon
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 1}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 2}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 3}
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Feedback Echo}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial echop0 {
@@ -458,7 +458,7 @@ refresh(eff);}
   } {
     Fl_Window effchoruswindow {
       label Chorus open
-      xywh {628 830 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      xywh {467 742 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -466,47 +466,47 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Chorus 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Chorus 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Chorus 3}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Celeste 1}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Celeste 2}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 1}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 2}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 3}
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 4}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 5}
-          xywh {110 110 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {110 110 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial chorusp0 {
@@ -593,8 +593,8 @@ refresh(eff);}
   Function {make_phaser_window()} {open
   } {
     Fl_Window effphaserwindow {
-      label Phaser selected
-      xywh {77 48 380 95} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      label Phaser open
+      xywh {101 232 380 95} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -602,55 +602,55 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 100 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 100 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Phaser 1}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 2}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 3}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 4}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 5}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 6}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 3}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 4}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 5}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {APhaser 6}
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial phaserp0 {
@@ -723,8 +723,8 @@ refresh(eff);}
       }
       Fl_Check_Button phaserp10 {
         label Substract
-        callback {eff->seteffectpar(10,(int) o->value());}
-        tooltip {inverts output} xywh {200 10 74 20} box THIN_UP_BOX down_box DOWN_BOX color 230 labelfont 1 labelsize 10
+        callback {eff->seteffectpar(10,(int) o->value());} selected
+        tooltip {inverts output} xywh {200 10 74 20} box THIN_UP_BOX down_box DOWN_BOX color 51 labelfont 1 labelsize 10
       }
       Fl_Dial phaserp11 {
         label Phase
@@ -754,7 +754,7 @@ refresh(eff);}
   } {
     Fl_Window effalienwahwindow {
       label AlienWah
-      xywh {715 462 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      xywh {253 353 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -762,23 +762,23 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Alienwah 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 3}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 4}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial awp0 {
@@ -860,8 +860,8 @@ refresh(eff);}
   Function {make_distorsion_window()} {open
   } {
     Fl_Window effdistorsionwindow {
-      label Distortion
-      xywh {695 261 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      label Distortion open
+      xywh {544 217 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -869,31 +869,31 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {11 15 95 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {11 15 95 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Overdrive 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Overdrive 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {A. Exciter 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {A. Exciter 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Guitar Amp}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Quantisize
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial distp0 {
@@ -1021,7 +1021,7 @@ refresh(eff);}
   } {
     Fl_Window effeqwindow {
       label EQ open
-      xywh {668 658 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      xywh {682 881 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1157,7 +1157,7 @@ eqgraph->redraw();}
   } {
     Fl_Window effdynamicfilterwindow {
       label DynFilter open
-      xywh {808 520 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
+      xywh {819 290 380 100} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 25
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1165,27 +1165,27 @@ eqgraph->redraw();}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 14 selection_color 0 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 14 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label WahWah
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label AutoWah
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Sweep
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label VocalMorph1
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label VocalMorph2
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial dfp0 {
@@ -1270,7 +1270,7 @@ refresh(eff);}
   } {
     Fl_Window filterwindow {
       label {Filter Parameters for DynFilter Eff.}
-      xywh {1454 437 290 110} type Double
+      xywh {801 474 290 110} type Double
       code0 {set_module_parameters(o);} visible
     } {
       Fl_Group {} {
@@ -1499,7 +1499,7 @@ effdynamicfilterwindow->hide();//delete (effdynamicfilterwindow);} {}
   } {
     Fl_Window effnullwindow {
       label {No Effect} open
-      xywh {1073 571 230 95} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 16
+      xywh {1047 755 230 95} type Double box UP_BOX color 221 labelfont 1 labelsize 19 align 16
       code0 {set_module_parameters(o);}
       class Fl_Group visible
     } {}
@@ -1508,7 +1508,7 @@ effdynamicfilterwindow->hide();//delete (effdynamicfilterwindow);} {}
   } {
     Fl_Window effreverbwindow {
       label Reverb open
-      xywh {1183 279 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {1047 463 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1517,59 +1517,59 @@ effdynamicfilterwindow->hide();//delete (effdynamicfilterwindow);} {}
         callback {eff->changepreset((int)o->value());
 
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Cathedral 1}
-          xywh {10 10 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {10 10 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Cathedral 2}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Cathedral 3}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Hall 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Hall 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Room 1}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Room 2}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Basement
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Tunnel
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echoed 1}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echoed 2}
-          xywh {110 110 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {110 110 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Very Long 1}
-          xywh {120 120 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {120 120 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Very Long 2}
-          xywh {130 130 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {130 130 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial revp0 {
@@ -1602,7 +1602,7 @@ refresh(eff);}
   } {
     Fl_Window effechowindow {
       label Echo open
-      xywh {404 639 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {428 823 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1610,43 +1610,43 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {11 15 95 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {11 15 95 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Echo 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echo 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Echo 3}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Simple Echo}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Canyon
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 1}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 2}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Panning Echo 3}
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Feedback Echo}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial echop0 {
@@ -1673,7 +1673,7 @@ refresh(eff);}
   } {
     Fl_Window effchoruswindow {
       label Chorus open
-      xywh {680 647 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {719 588 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1681,47 +1681,47 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Chorus 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Chorus 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Chorus 3}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Celeste 1}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Celeste 2}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 1}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 2}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 3}
-          xywh {90 90 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {90 90 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 4}
-          xywh {100 100 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {100 100 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Flange 5}
-          xywh {110 110 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {110 110 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial chorusp0 {
@@ -1766,7 +1766,7 @@ refresh(eff);}
   } {
     Fl_Window effphaserwindow {
       label Phaser open
-      xywh {1485 647 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {1047 831 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1774,31 +1774,31 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
       } {
         MenuItem {} {
           label {Phaser 1}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 2}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 3}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 4}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 5}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Phaser 6}
-          xywh {80 80 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {80 80 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial phaserp0 {
@@ -1843,7 +1843,7 @@ refresh(eff);}
   } {
     Fl_Window effalienwahwindow {
       label AlienWah open
-      xywh {379 296 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {403 480 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1851,23 +1851,23 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Alienwah 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 3}
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Alienwah 4}
-          xywh {70 70 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {70 70 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial awp0 {
@@ -1900,7 +1900,7 @@ refresh(eff);}
   } {
     Fl_Window effdistorsionwindow {
       label Distortion open
-      xywh {456 797 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {353 881 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -1908,31 +1908,31 @@ refresh(eff);}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {11 15 95 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {11 15 95 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label {Overdrive 1}
-          xywh {20 20 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {20 20 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Overdrive 2}
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {A. Exciter 1}
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {A. Exciter 2}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label {Guitar Amp}
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Quantisize
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial distp0 {
@@ -2027,7 +2027,7 @@ refresh(eff);}
   } {
     Fl_Window effeqwindow {
       label EQ open
-      xywh {1366 827 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {1047 881 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -2156,7 +2156,7 @@ eqgraph->redraw();}
   } {
     Fl_Window effdynamicfilterwindow {
       label DynFilter open
-      xywh {1347 465 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
+      xywh {965 527 230 100} type Double box UP_BOX color 51 labelfont 1 labelsize 19 align 25
       code3 {set_module_parameters(o);}
       class Fl_Group visible
     } {
@@ -2164,27 +2164,27 @@ eqgraph->redraw();}
         label Preset
         callback {eff->changepreset((int)o->value());
 refresh(eff);}
-        xywh {10 15 90 15} down_box BORDER_BOX color 47 selection_color 0 labelfont 1 labelsize 10 labelcolor 55 align 5 textfont 1 textsize 10 textcolor 7
+        xywh {10 15 90 15} box UP_BOX down_box BORDER_BOX color 47 selection_color 7 labelfont 1 labelsize 10 align 5 textfont 1 textsize 10
       } {
         MenuItem {} {
           label WahWah
-          xywh {30 30 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {30 30 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label AutoWah
-          xywh {40 40 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {40 40 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label Sweep
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label VocalMorph1
-          xywh {50 50 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {50 50 100 20} labelfont 1 labelsize 10
         }
         MenuItem {} {
           label VocalMorph2
-          xywh {60 60 100 20} labelfont 1 labelsize 10 labelcolor 7
+          xywh {60 60 100 20} labelfont 1 labelsize 10
         }
       }
       Fl_Dial dfp0 {
diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index 6695a318..29b338ec 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -324,7 +324,7 @@ class Panellistitem {open : {public Fl_Group}
   Function {make_window()} {open private
   } {
     Fl_Window panellistitem {open
-      private xywh {602 665 100 260} type Double box NO_BOX
+      private xywh {605 688 100 260} type Double box NO_BOX
       class Fl_Group visible
     } {
       Fl_Group panellistitemgroup {open
@@ -463,7 +463,7 @@ if ((
     *exitprogram=1;
 };
 \#endif} open
-      xywh {623 378 390 545} type Double align 80 xclass zynaddsubfx visible
+      xywh {77 347 390 525} type Double align 80 xclass zynaddsubfx visible
     } {
       Fl_Menu_Bar mastermenu {
         xywh {-5 0 690 25}
@@ -728,10 +728,10 @@ pthread_mutex_unlock(&master->mutex);}
         xywh {280 29 105 53} color 90 labelfont 1
       }
       Fl_Group partuigroup {open
-        xywh {0 332 390 183}
+        xywh {0 310 390 205}
       } {
-        Fl_Group partui {open
-          xywh {4 335 383 175}
+        Fl_Group partui {open selected
+          xywh {0 310 383 175}
           code0 {o->init(master->part[0],master,0,bankui);}
           code1 {o->show();}
           class PartUI
@@ -987,11 +987,11 @@ mastermenu->redraw();}
         }
       }
       Fl_Group {} {open
-        xywh {1 512 389 33}
+        xywh {1 490 389 55}
       } {
         Fl_Box {} {
           label {VU-Meter}
-          xywh {5 515 380 30} box FLAT_BOX color 48 selection_color 75
+          xywh {5 490 380 30} box FLAT_BOX color 48 selection_color 75
           code0 {o->init(master,-1);}
           class VUMeter
         }
@@ -1141,7 +1141,7 @@ if (fl_choice("Exit and leave the unsaved data?","No","Yes",NULL)) {
 	*exitprogram=1;
 };
 \#endif} open
-      xywh {277 216 600 335} type Double align 80 visible
+      xywh {280 239 600 335} type Double align 80 visible
     } {
       Fl_Menu_Bar simplemastermenu {
         xywh {0 0 690 25}
@@ -1331,7 +1331,7 @@ simplemaxkcounter->do_callback();}
           code1 {o->value(master->part[npart]->Pkeyshift-64);}
         }
         Fl_Dial simplesyseffsend {
-          callback {master->setPsysefxvol(npart,nsyseff,(int) o->value());} selected
+          callback {master->setPsysefxvol(npart,nsyseff,(int) o->value());}
           xywh {300 160 30 30} maximum 127 step 1
           class WidgetPDial
         }
diff --git a/src/UI/PADnoteUI.fl b/src/UI/PADnoteUI.fl
index 25c0e32f..aebc11c3 100644
--- a/src/UI/PADnoteUI.fl
+++ b/src/UI/PADnoteUI.fl
@@ -160,10 +160,10 @@ const int maxharmonic=64;
 
 
 for (int i=1;i<maxharmonic;i++){
-	fl_color(200,200,200);
+	fl_color(100,100,100);
 	fl_line_style(FL_DOT);
 	if (i%5==0) fl_line_style(0);
-	if (i%10==0) fl_color(160,160,160);
+	if (i%10==0) fl_color(120,120,120);
 	int kx=(int)(lx/(float)maxharmonic*i);
 	fl_line(ox+kx,oy,ox+kx,oy+ly);
 };
diff --git a/src/UI/VirKeyboard.fl b/src/UI/VirKeyboard.fl
index 1a0c6711..596dcb07 100644
--- a/src/UI/VirKeyboard.fl
+++ b/src/UI/VirKeyboard.fl
@@ -95,6 +95,15 @@ rndvelocity=0;} {}
  Fl_Image *black_up = Fl_Shared_Image::get( PIXMAP_PATH "black_key.png" );
  Fl_Image *black_down = Fl_Shared_Image::get( PIXMAP_PATH "black_key_pressed.png" );
 
+
+//On error fetch everything from source directory
+if(!(white_up&&white_down&&black_up&&black_down)) {
+    white_up   = Fl_Shared_Image::get(SOURCE_DIR "/../../pixmaps/white_key.png");   
+    white_down = Fl_Shared_Image::get(SOURCE_DIR "/../../pixmaps/white_key_pressed.png");
+    black_up   = Fl_Shared_Image::get(SOURCE_DIR "/../../pixmaps/black_key.png");
+    black_down = Fl_Shared_Image::get(SOURCE_DIR "/../../pixmaps/black_key_pressed.png");
+}
+
 Fl_Image *key;
 
 for (i=0;i<N_OCT*12;i++) {
diff --git a/src/main.cpp b/src/main.cpp
index 7e1ac430..0ebde8bf 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -112,7 +112,7 @@ set_module_parameters ( Fl_Widget *o )
     o->image( module_backdrop );
     o->labeltype( FL_SHADOW_LABEL );
 #else
-    o->box( FL_UP_BOX );
+    o->box( FL_PLASTIC_UP_BOX );
     o->color( FL_CYAN );
     o->labeltype( FL_EMBOSSED_LABEL );
 #endif
@@ -435,11 +435,22 @@ int main(int argc, char *argv[])
 #ifdef NTK_GUI
     fl_register_images();
 
-    Fl_Dial::default_style( Fl_Dial::PIXMAP_DIAL );
-    Fl_Dial::default_image( Fl_Shared_Image::get( PIXMAP_PATH "/knob.png" ) );
+    Fl_Dial::default_style(Fl_Dial::PIXMAP_DIAL);
 
-    Fl::scheme_bg( new Fl_Tiled_Image( Fl_Shared_Image::get( PIXMAP_PATH "/window_backdrop.png" )));
-    module_backdrop = new Fl_Tiled_Image( Fl_Shared_Image::get( PIXMAP_PATH "/module_backdrop.png" ));
+    if(Fl_Shared_Image *img = Fl_Shared_Image::get(PIXMAP_PATH "/knob.png"))
+        Fl_Dial::default_image(img);
+    else
+        Fl_Dial::default_image(Fl_Shared_Image::get(SOURCE_DIR "/../pixmaps/knob.png"));
+
+    if(Fl_Shared_Image *img = Fl_Shared_Image::get(PIXMAP_PATH "/window_backdrop.png"))
+        Fl::scheme_bg(new Fl_Tiled_Image(img));
+    else
+        Fl::scheme_bg(new Fl_Tiled_Image(Fl_Shared_Image::get(SOURCE_DIR "/../pixmaps/window_backdrop.png")));
+
+    if(Fl_Shared_Image *img = Fl_Shared_Image::get(PIXMAP_PATH "/module_backdrop.png"))
+        module_backdrop = new Fl_Tiled_Image(img);
+    else
+        module_backdrop = new Fl_Tiled_Image(Fl_Shared_Image::get(SOURCE_DIR "/../pixmaps/module_backdrop.png"));
 
     Fl::background(  50, 50, 50 );
     Fl::background2(  70, 70, 70 );
@@ -521,7 +532,7 @@ int main(int argc, char *argv[])
 
 done:
 
-        Fl::wait(0.1f);
+        Fl::wait(0.02f);
 #else
         usleep(100000);
 #endif
-- 
2.47.0

